src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -1026,7 +1032,7 @@ private TypedConstant VisitExpression(BoundExpression node, DiagnosticBag diagno
             {
                 // Validate Statement 1) of the spec comment above.
 
-                var typedConstantKind = node.Type.GetAttributeParameterTypedConstantKind(_binder.Compilation);
```

Comment:
Given `node.Type` is external to the method consider using `RoslynDebug.Assert` for this.